Index: branches/1.3-stable/licq/src/hebrev.c
===================================================================
--- licq/src/hebrev.c (revision 6819)
+++ licq/src/hebrev.c (revision 7000)
@@ -7,7 +7,5 @@
  */
 
-#ifdef HAVE_CONFIG_H
 #include "config.h"
-#endif
 
 #ifdef USE_HEBREW
@@ -49,4 +47,5 @@
 	const char *hmark = NULL, *lmark, *nmark, *nlmark;
 	char ch;
+  const char* srcstart = src;
 
 	if (src == NULL)
@@ -70,5 +69,6 @@
 			{
 				lmark = src-1;
-				while ((!isheb(*lmark)) && (!ispunct(*lmark))) lmark--;
+        while (lmark > srcstart && !isheb(*lmark) && !ispunct(*lmark))
+          lmark--;
 				src = lmark;
 				imode = 0;
@@ -115,4 +115,6 @@
 				mode = 0;
 			}
+      if (*src == '\0')
+        *dest = '\0';
 		}
 		if (!*src++)
@@ -142,4 +144,5 @@
 	char *arg=NULL, *arg2=NULL;
 	int i=0;
+  char* saveptr = NULL;
 
 	temp = (char*) malloc(strlen(input)+1);
@@ -148,7 +151,7 @@
 	strcpy(temp, input);
 
-	arg = strtok(temp, "\n");
+  arg = strtok_r(temp, "\n", &saveptr);
 	for(i = 0; (i < index) && (arg!=NULL); i++)
-		arg = strtok(NULL, "\n");
+    arg = strtok_r(NULL, "\n", &saveptr);
 
 	if(arg != NULL)
@@ -165,4 +168,5 @@
         char* temp_str = NULL;
 	char* temp = NULL;
+  char* arg = NULL;
         int i=0;
 	int size = 0;
@@ -173,9 +177,10 @@
 		return NULL;
 
-	temp = GetArg(src, i);
-	while(temp != NULL)
+  arg = GetArg(src, i);
+	while(arg != NULL)
 	{
 		i++;
-		temp = hebrew(temp);
+  temp = hebrew(arg);
+  free(arg);
 		memcpy(temp_str + size , temp, strlen(temp));
 		size += strlen(temp) ;
@@ -184,8 +189,9 @@
 		free(temp);
 		temp = NULL;
-		temp = GetArg(src, i);
+    arg = GetArg(src, i);
 	}
-	free(temp);
-	temp_str[size]= '\0';	
+  // Above loop adds a line break after last line that wasn't there in src
+  //   replace it with the null terminator
+  temp_str[--size]= '\0';
 	return temp_str;
 }
Index: branches/1.3-stable/licq/src/user.cpp
===================================================================
--- licq/src/user.cpp (revision 6819)
+++ licq/src/user.cpp (revision 7000)
@@ -3641,4 +3641,5 @@
       gLog.Error("%sUnable to open picture file (%s):\n%s%s.\n", L_ERRORxSTR,
                                      szFilename, L_BLANKxSTR, strerror(errno));
+      close(source);
       return;
     }
Index: branches/1.3-stable/licq/src/icqd-srv.cpp
===================================================================
--- licq/src/icqd-srv.cpp (revision 6819)
+++ licq/src/icqd-srv.cpp (revision 7000)
@@ -307,5 +307,5 @@
   gUserManager.DropUser(u);
 
-  if (ppid == LICQ_PPID && tempUser)
+  if (ppid == LICQ_PPID && !tempUser)
     icqRemoveUser(accountId.c_str());
   else if(ppid != LICQ_PPID)
