Index: /trunk/licq/src/user.cpp
===================================================================
--- licq/src/user.cpp (revision 6943)
+++ licq/src/user.cpp (revision 6978)
@@ -2974,5 +2974,5 @@
   // TODO Make the protocol plugin normalize the accountId
   // For AIM, account id is case insensitive and spaces should be ignored
-  if (ppid == LICQ_PPID && !isdigit(accountId[0]))
+  if (ppid == LICQ_PPID && !accountId.empty() && !isdigit(accountId[0]))
   {
     boost::erase_all(realId, " ");
Index: /trunk/qt4-gui/src/config/iconmanager.cpp
===================================================================
--- licq/plugins/qt4-gui/src/config/iconmanager.cpp (revision 6899)
+++ licq/plugins/qt4-gui/src/config/iconmanager.cpp (revision 6978)
@@ -282,5 +282,5 @@
 const QPixmap& IconManager::iconForStatus(unsigned long fullStatus, const QString& id, unsigned long ppid)
 {
-  bool isAim = (ppid == LICQ_PPID) && (!id[0].isDigit());
+  bool isAim = (ppid == LICQ_PPID) && !id.isEmpty() && (!id[0].isDigit());
 
   ProtocolType protocol = (isAim ? ProtocolAim : static_cast<ProtocolType>(ppid));
