Index: branches/1.3-stable/licq/src/socket.cpp
===================================================================
--- licq/src/socket.cpp (revision 7023)
+++ licq/src/socket.cpp (revision 7087)
@@ -501,5 +501,5 @@
 
 #ifdef IPV6_PORTRANGE
-    int i = IPV6_PORTRANGE_HIGH;
+    i = IPV6_PORTRANGE_HIGH;
     if (setsockopt(m_nDescriptor, IPPROTO_IPV6, IPV6_PORTRANGE, &i, sizeof(i)) < 0)
     {
