Subject: Adjusts some configuration defaults to be more reasonable for Debian
Origin: vendor, Debian package
Index: licq/src/daemon.cpp
===================================================================
--- licq.orig/src/daemon.cpp	2010-09-23 22:06:20.000000000 +0200
+++ licq/src/daemon.cpp	2010-09-23 22:08:14.000000000 +0200
@@ -131,7 +131,7 @@
     Licq::gTranslator.setTranslationMap(shareDir() + TranslationDir + "/" + temp);
 
   // Misc
-  licqConf.get("Terminal", myTerminal, "xterm -T Licq -e ");
+  licqConf.get("Terminal", myTerminal, "x-terminal-emulator -T Licq -e ");
   licqConf.get("SendTypingNotification", mySendTypingNotification, true);
   licqConf.get("IgnoreTypes", myIgnoreTypes, 0);
 
