Index: licq/plugins/qt-gui/configure
===================================================================
--- licq.orig/plugins/qt-gui/configure	2010-05-18 21:10:19.000000000 +0200
+++ licq/plugins/qt-gui/configure	2010-07-17 15:01:15.000000000 +0200
@@ -18768,7 +18768,7 @@
 
 
 
-  if test "$have_kde" == "yes"; then
+  if test "$have_kde" = "yes"; then
     # Extract the first word of "dcopidl", so it can be a program name with args.
 set dummy dcopidl; ac_word=$2
 { $as_echo "$as_me:${as_lineno-$LINENO}: checking for $ac_word" >&5
