commit e2fefa9d05cd8d0968039012c9c92339eeed21b5
Author: Erik Johansson <erik@ejohansson.se>
Date:   Thu Apr 7 21:08:34 2011 +0200

    Link with thread library the normal way

diff --git a/licq/cmake/FindLicq.cmake b/licq/cmake/FindLicq.cmake
index daf5f61..4dc700a 100644
--- licq/cmake/FindLicq.cmake
+++ licq/cmake/FindLicq.cmake
@@ -84,11 +84,11 @@ macro (LICQ_ADD_PLUGIN _licq_plugin_name)
   add_library(${_licq_plugin_name} MODULE ${ARGN})
   set_target_properties(${_licq_plugin_name} PROPERTIES PREFIX "")
 
-  # Add linker flags for thread support
-  set(_link_flags "${CMAKE_THREAD_LIBS_INIT}")
+  # Link with thread library
+  target_link_libraries(${_licq_plugin_name} ${CMAKE_THREAD_LIBS_INIT})
 
   if (APPLE)
-    set(_link_flags "${_link_flags} -flat_namespace -undefined suppress")
+    set(_link_flags "-flat_namespace -undefined suppress")
 
     # Write the list of symbols that should be exported from the plugin to a
     # file and tell the linker about it. One symbol per line with a '_' prefix.
@@ -109,8 +109,7 @@ macro (LICQ_ADD_PLUGIN _licq_plugin_name)
     check_cxx_accepts_flag("-Wl,--version-script,${_version_script}"
       LD_ACCEPTS_VERSION_SCRIPT)
     if (LD_ACCEPTS_VERSION_SCRIPT)
-      set(_link_flags
-	"${_link_flags} -Wl,--version-script,'${_version_script}'")
+      set(_link_flags "-Wl,--version-script,'${_version_script}'")
     endif (LD_ACCEPTS_VERSION_SCRIPT)
   endif (APPLE)
 
diff --git a/licq/src/CMakeLists.txt b/licq/src/CMakeLists.txt
index 7e3b214..7b21f80 100644
--- licq/src/CMakeLists.txt
+++ licq/src/CMakeLists.txt
@@ -110,8 +110,8 @@ set(test_SRCS
 
 add_executable(licq ${licq_SRCS})
 
-# Add linker flags for thread support
-set_target_properties(licq PROPERTIES LINK_FLAGS "${CMAKE_THREAD_LIBS_INIT}")
+# Link with thread library
+target_link_libraries(licq ${CMAKE_THREAD_LIBS_INIT})
 
 # Must export symbols from daemon so that plugins can use them
 set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${CMAKE_EXE_EXPORTS_CXX_FLAG}")
@@ -152,9 +152,8 @@ if (BUILD_TESTS)
   target_link_libraries(unittests ${CMAKE_DL_LIBS})
   target_link_libraries(unittests ${SOCKET_LIBRARIES})
 
-  # Add linker flags for thread support
-  set_target_properties(unittests PROPERTIES
-    LINK_FLAGS "${CMAKE_THREAD_LIBS_INIT}")
+  # Link with thread library
+  target_link_libraries(unittests ${CMAKE_THREAD_LIBS_INIT})
 
   add_test(licq unittests)
   add_dependencies(tests unittests)
